main/ejml-dsparse/src/org/ejml/sparse/csc/RandomMatrices_DSCC.java

```diff
@@ -277,6 +278,62 @@ public static DMatrixSparseCSC symmetricPosDef( int width, double probabilityZer
         return out;
     }
 
+    /**
+     * Creates a random matrix where each column has exactly `nzEntriesPerColumn` non-zero entries.
+     * Compared to `rectangle` this method can generate larger sparse matrices.
+     *
+     * @param numRows Number of rows
+     * @param numCols Number of columns
+     * @param nzEntriesPerColumn Amount of nz-entries per column
+     * @param min Minimum element value, inclusive
+     * @param max Maximum element value, inclusive
+     * @param sorted Whether the nz-entries should be sorted
+     * @param rand Random number generator
+     * @return Randomly generated matrix
+     */
+    public static DMatrixSparseCSC generateUniform(int numCols, int numRows, int nzEntriesPerColumn,
+                                                   double min, double max, boolean sorted, Random rand) {
+        if (nzEntriesPerColumn > numRows) {
+            throw new IllegalArgumentException("numRows must be greater than nzEntriesPerColumn");
+        }
+
+        int nz_total = Math.toIntExact(nzEntriesPerColumn * numCols);
+
+        DMatrixSparseCSC matrix = new DMatrixSparseCSC(numCols, numRows, nz_total);
+        matrix.indicesSorted = sorted;
+
+
+        int[] nz_hist = new int[numCols];
+        Arrays.fill(nz_hist, nzEntriesPerColumn);
+        matrix.histogramToStructure(nz_hist);
+
+        BitSet existingRows = new BitSet(numRows);
+
+        for (int i = 0; i < nz_total; i++) {
+            if (i % nzEntriesPerColumn == 0) {
```

Comment:
Will this cause it to have a triangle like pattern along the diagonal? Just curious. I'm also considering reviving the code on visualizing matrices.

Reply:
No this is just the reset per column, as previous entries do not matter. I could have also wrote it by iterating per column, but this way I easily got the corresponding entry in the nz arrays.